tests/Codec/LazyBSONDocumentCodecTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use MongoDB\BSON\Document;
+use MongoDB\Codec\LazyBSONDocumentCodec;
+use MongoDB\Exception\UnsupportedValueException;
+use MongoDB\Model\LazyBSONDocument;
+use MongoDB\Tests\TestCase;
+
+class LazyBSONDocumentCodecTest extends TestCase
+{
+    private const OBJECT = [
+        'foo' => 'bar',
+        'document' => ['bar' => 'baz'],
+        'array' => [0, 1, 2],
+    ];
+
+    public function testDecode(): void
+    {
+        $document = (new LazyBSONDocumentCodec())->decode($this->getTestDocument());
+
+        $this->assertInstanceOf(LazyBSONDocument::class, $document);
+        $this->assertSame('bar', $document->foo);
+    }
+
+    public function testDecodeWithWrongType(): void
+    {
+        $codec = new LazyBSONDocumentCodec();
+
+        $this->expectExceptionObject(UnsupportedValueException::invalidDecodableValue('foo'));
+        $codec->decode('foo');
+    }
+
+    public function testEncode(): void
+    {
+        $document = new LazyBSONDocument($this->getTestDocument());
+        $encoded = (new LazyBSONDocumentCodec())->encode($document);
```

Comment:
Noted that `assertInstanceOf()` was omitted here as well. Was that intentional?

Reply:
Added.